app/models/course_data/articles_courses.rb

```diff
@@ -132,7 +131,12 @@ def self.destroy_invalid_records(course, valid_article_ids)
     end
   end
 
-  def self.get_revisions_by_namespaces(namespaces, revisions)
-    revisions.joins(:article).where(articles: { namespace: namespaces })
+  def self.get_revisions_by_namespaces(course)
+    # Return revisions corresponding to tracked wikis and namespaces
+    course.tracked_namespaces.map do |wiki_ns|
+      wiki = wiki_ns[:wiki]
+      namespace = wiki_ns[:namespace]
+      course.revisions.joins(:article).where(articles: { wiki: wiki, namespace: namespace })
+    end.flatten
```

Comment:
I think this is going to be a problem, because it's creating an array of all the revisions in a course. It's likely to run the system out of memory when it does this for a course with a very large number of revisions. We only actually need the `article_id` values, so it would be better to change this method to return an array of ids directly; a large array of integers takes up a trivial amount of memory compared to the same number of ActiveRecord objects.

Reply:
Oh...right, got it.

Reply:
I'm not certain, but I don't think so. An ActiveRecord Relation object doesn't create an array of ActiveRecord objects, unless it gets converted to an array or iterated through. So in the current version, that Relation object gets `.distinct.pluck(:article_id)` called on it, which just returns the array of ids without individually creating an Article or Revision object for each record. Using Enumerable iterator methods `map` and/or `flatten`, on the other hand, do create an array of such objects.